### ISSUE TITLE
DIY Freenove Wrover QR read error

### ISSUE THREAD

**Issue by joksa** [Blockstream/Jade#176]
I built Jade on Freenove ESP32 Wrover
The camera module is OV3660 (need to manually check in the menuconfig)

 On QR scanning I got an error in Log console
clogX?I (1519171) ./main/qrscan.c: 29: Detected 1 QR codes in image.
clogX9W (1519171) ./main/qrscan.c: 42: QUIRC error ECC failure

Install is from the latest source.
I appreciate any help on this matter.

**Comment by JamieDriver** [comment]
That error is quite common.   
The camera is taking several frames per second and passing them to the QR-reader.  There are many frames which are initially recognised as containing a QR code, but then full analysis errors - maybe because of motion blur, focus issues, QR too close or too far away, or maybe because the display was part-way through updating the QR (animated/multi-frame QR); any of which can result in this error.
Usually just holding the camera steady, or slowly moving the QR closer or further will mean that subsequent captures will correctly read the QR code.

I don't know about that board or camera specifically - but you might need to experiment to work out the ideal size/position of the QR being scanned.

**Comment by JamieDriver** [comment]
Did you manage to get QR scanning to work ?

**Comment by joksa** [comment]
Unfortunately no. I hope that CryptoGuide will also test it when he receives the same hardware, until then the hardware is on hold.

**Comment by JamieDriver** [comment]
Yes, there are plenty of camera modules around -  we have concentrated on GC0308 as that's the one we found to be the most reliable of the ones we tried, and the one we chose to use in our production Jade units.
It may be that some tweaks can be made for other cameras to adjust contrast, brightness/exposure etc.
See also #190

**Comment by gesooNakamoto** [comment]
I also have the same problem with the same camera model. Were you able to solve it?

**Comment by caiocfer** [comment]
I was able to solve it by flipping the camera vertically, I submitted a PR, feel free to try it :)

https://github.com/Blockstream/Jade/pull/218

**Comment by JamieDriver** [comment]
Merged @caiocfer's fix to master at 7a8de016 - does this resolve the issues with this camera model ?
